refinery/data_set_manager/tests.py

```diff
@@ -361,14 +363,181 @@ def test_put_invalid_login(self):
+
+        assay = Assay.objects.create(
+                study=study,
+                measurement='transcription factor binding site',
+                measurement_accession='http://www.testurl.org/testID',
+                measurement_source='OBI',
+                technology='nucleotide sequencing',
+                technology_accession='test info',
+                technology_source='test source',
+                platform='Genome Analyzer II',
+                file_name='test_assay_filename.txt',
+                )
+        self.valid_uuid = assay.uuid
+        self.invalid_uuid = "0xxx000x-00xx-000x-xx00-x00x00x00x0x"
+        self.url = "/api/v2/assays/%s/files/"
+        self.non_meta_attributes = ['REFINERY_DOWNLOAD_URL', 'REFINERY_NAME']
+        self.client = APIClient()
+
+    def tearDown(self):
+        Assay.objects.all().delete()
+        Study.objects.all().delete()
+        InvestigationLink.objects.all().delete()
+        Investigation.objects.all().delete()
+        DataSet.objects.all().delete()
+
+    @mock.patch('data_set_manager.views.generate_solr_params_for_assay')
+    @mock.patch('data_set_manager.views.search_solr')
+    @mock.patch('data_set_manager.views.format_solr_response')
+    def test_get_from_owner_with_valid_params(self,
+                                              mock_format,
+                                              mock_search,
+                                              mock_generate):
+        self.client.login(username=self.user_owner,
+                          password=self.fake_password)
+        mock_format.return_value = {'status': 200}
+        uuid = self.valid_uuid
+        params = {
+            'limit': '0',
+            'data_set_uuid': self.data_set.uuid
+        }
+        response = self.client.get(self.url % uuid, params)
+        self.assertTrue(mock_format.called)
+        self.assertTrue(mock_search.called)
+        qdict = QueryDict('', mutable=True)
+        qdict.update(params)
+        mock_generate.assert_called_once_with(qdict, uuid)
+        self.assertEqual(response.status_code, 200)
+
+    def test_get_from_owner_invalid_params(self):
+        self.client.login(username=self.user_owner,
+                          password=self.fake_password)
+
+        uuid = self.valid_uuid
+        params = {
+            'limit': 0,
+            'data_set_uuid': self.invalid_uuid
+        }
+        response = self.client.get(self.url % uuid, params)
+        self.assertEqual(response.status_code, 404)
+        self.client.logout()
+
+    @mock.patch('data_set_manager.views.generate_solr_params_for_assay')
+    @mock.patch('data_set_manager.views.search_solr')
+    @mock.patch('data_set_manager.views.format_solr_response')
+    def test_get_from_user_no_perms(self,
+                                    mock_format,
+                                    mock_search,
+                                    mock_generate):
+        self.client.login(username=self.user_guest,
```

Comment:
Hw about having the login and logout portions for these tests in a `setUp()` and `tearDown()` respectively?